### PR TITLE
Got rid of `S3AuthSigner`

### PR DESCRIPTION
**Pull request by excitoon**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

It will allow to remove ClickHouse/aws-sdk-cpp#3.
FYI @Jokser

### CODE REVIEW

**Comment by Jokser** [comment]
For me, it looks good.
But it should be probably also reviewed by @GrigoryPervakov

File: src/IO/S3Common.cpp [hunk 1]
```diff
@@ -747,17 +679,20 @@ namespace S3
...
-            client_configuration,
-            std::move(credentials),
-            std::move(headers),
-            use_environment_credentials,
-            use_insecure_imds_request);
+        client_configuration.extra_headers = std::move(headers);
+
+        Aws::Auth::AWSCredentials credentials(access_key_id, secret_access_key);
+        auto credentials_provider = std::make_shared<S3CredentialsProviderChain>(
+                static_cast<const DB::S3::PocoHTTPClientConfiguration &>(client_configuration),
```
[the remark this answers is not in the document]
Comment (excitoon) [reply src/IO/S3Common.cpp:686]: You are right.

[the remark this answers is not in the document]
Comment (excitoon) [reply src/IO/S3Common.cpp:686]: But actually `S3CredentialsProviderChain` uses quite a lot from `client_configuration`, to create its own clients to access some credentials, in particular.

**Comment by excitoon** [comment]
@nikitamikhaylov what shall be the next steps?

**Comment by nikitamikhaylov** [comment]
@excitoon Please fix the build

File: src/IO/S3/tests/TestPocoHTTPServer.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        server_params(new Poco::Net::HTTPServerParams())
+    {
+        std::random_device device;
+        std::mt19937 generator(device());
+        std::uniform_int_distribution<> distribution(10000, 60000);
+        for (int attempt = 100; attempt >= 0; --attempt)
+        {
+            try
+            {
+                server_socket = std::make_unique<Poco::Net::ServerSocket>(distribution(generator));
```
Comment (nikitamikhaylov) [inline src/IO/S3/tests/TestPocoHTTPServer.h:52]: If you pass 0 as port, the port will be determined automatically

**Review by nikitamikhaylov — APPROVED** [review]

File: src/IO/S3/tests/TestPocoHTTPServer.h (hunk 2 above)
Comment (excitoon) [reply src/IO/S3/tests/TestPocoHTTPServer.h:52]: Yeah, it works.

**Comment by excitoon** [comment]
| test_keeper_zookeeper_converter/test.py::test_simple_crud_requests[False] | FAIL | 28.48 |
| test_keeper_zookeeper_converter/test.py::test_multi_and_failed_requests[False] | FAIL | 24.52 |

**Review by nikitamikhaylov — APPROVED** [review]

**Pull request merged**